src/serialize.cpp

```diff
@@ -75,9 +75,11 @@ void migraphx_from_value(const value& v, literal& l)
     else
     {
         auto v_data = v.at("data");
-        assert(v_data.is_array());
+        if(not v_data.is_array())
+        {
+            MIGRAPHX_THROW("Literal is larger than 4GB but it is not stored as binary array");
```

Comment:
If you want to add exceptions for this, this should go into msgpack. Other formats(such as json) do not have these limitations.

Reply:
Would anyone use JSON for such large data ? It would be too slow.